apps/portal/src/server/router/questions-question-encounter-router.ts

```diff
@@ -96,14 +118,48 @@ export const questionsQuestionEncounterRouter = createProtectedRouter()
         });
       }
 
-      return await ctx.prisma.questionsQuestionEncounter.update({
-        data: {
-          ...input,
-        },
-        where: {
-          id: input.id,
-        },
+      return await ctx.prisma.$transaction(async (tx) => {
+        const questionToUpdate = await tx.questionsQuestion.findUnique({
```

Comment:
Can we execute this `await` and the following one in parallel?

Reply:
Believe we can. Dont think the R and W have happens-before/after relationship since we can decide what we want to do with the question any time after the encounter write happens so the question read can happen anytime (synchronized at L137).

Reply:
Don't think so, let's parallelize them then!